src/node_http2.cc

```diff
@@ -1364,16 +1364,34 @@ void Http2Session::MaybeScheduleWrite() {
 // storage for data and metadata that was associated with these writes.
 void Http2Session::ClearOutgoing(int status) {
   CHECK_NE(flags_ & SESSION_STATE_SENDING, 0);
-  flags_ &= ~SESSION_STATE_SENDING;
 
-  for (const nghttp2_stream_write& wr : outgoing_buffers_) {
-    WriteWrap* wrap = wr.req_wrap;
-    if (wrap != nullptr)
-      wrap->Done(status);
+  if (outgoing_buffers_.size() > 0) {
+    for (const nghttp2_stream_write& wr : outgoing_buffers_) {
+      WriteWrap* wrap = wr.req_wrap;
+      if (wrap != nullptr)
+        wrap->Done(status);
+    }
+
+    outgoing_buffers_.clear();
+    outgoing_storage_.clear();
   }
 
-  outgoing_buffers_.clear();
-  outgoing_storage_.clear();
+  flags_ &= ~SESSION_STATE_SENDING;
```

Comment:
As far as I know, out streams implementation would typically send data that has buffered up on the writable side from within the `wrap->Done()` callback… it’s probably worth checking the benchmarks here

Reply:
Ok, I managed to run these on a separate machine on DO. No regressions that I could see...